var CapturedExceptions = {};

CapturedExceptions.opera_854 = {
    message: "Statement on line 44: Type mismatch (usually a non-object value used where an object is required)\n" +
        "Backtrace:\n" +
        "  Line 44 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    this.undef();\n" +
        "  Line 31 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    ex = ex || this.createException();\n" +
        "  Line 18 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "  Line 4 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    printTrace(printStackTrace());\n" +
        "  Line 7 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    bar(n - 1);\n" +
        "  Line 11 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    bar(2);\n" +
        "  Line 15 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    foo();\n" +
        "",
    'opera#sourceloc': 44
};

CapturedExceptions.opera_902 = {
    message: "Statement on line 44: Type mismatch (usually a non-object value used where an object is required)\n" +
        "Backtrace:\n" +
        "  Line 44 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    this.undef();\n" +
        "  Line 31 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    ex = ex || this.createException();\n" +
        "  Line 18 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "  Line 4 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    printTrace(printStackTrace());\n" +
        "  Line 7 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    bar(n - 1);\n" +
        "  Line 11 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    bar(2);\n" +
        "  Line 15 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    foo();\n" +
        "",
    'opera#sourceloc': 44
};

CapturedExceptions.opera_927 = {
    message: "Statement on line 43: Type mismatch (usually a non-object value used where an object is required)\n" +
        "Backtrace:\n" +
        "  Line 43 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    this.undef();\n" +
        "  Line 31 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    ex = ex || this.createException();\n" +
        "  Line 18 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "    var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "  Line 4 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    printTrace(printStackTrace());\n" +
        "  Line 7 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    bar(n - 1);\n" +
        "  Line 11 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    bar(2);\n" +
        "  Line 15 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "    foo();\n" +
        "",
    'opera#sourceloc': 43
};

CapturedExceptions.opera_964 = {
    message: "Statement on line 42: Type mismatch (usually non-object value supplied where object required)\n" +
        "Backtrace:\n" +
        "  Line 42 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "                this.undef();\n" +
        "  Line 27 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "            ex = ex || this.createException();\n" +
        "  Line 18 of linked script file://localhost/G:/js/stacktrace.js: In function printStackTrace\n" +
        "        var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "  Line 4 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function bar\n" +
        "             printTrace(printStackTrace());\n" +
        "  Line 7 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function bar\n" +
        "           bar(n - 1);\n" +
        "  Line 11 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function foo\n" +
        "           bar(2);\n" +
        "  Line 15 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "         foo();\n" +
        "",
    'opera#sourceloc': 42,
    stacktrace: "  ...  Line 27 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "            ex = ex || this.createException();\n" +
        "  Line 18 of linked script file://localhost/G:/js/stacktrace.js: In function printStackTrace\n" +
        "        var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "  Line 4 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function bar\n" +
        "             printTrace(printStackTrace());\n" +
        "  Line 7 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function bar\n" +
        "           bar(n - 1);\n" +
        "  Line 11 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function foo\n" +
        "           bar(2);\n" +
        "  Line 15 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "         foo();\n" +
        ""
};

CapturedExceptions.opera_1010 = {
    message: "Statement on line 42: Type mismatch (usually non-object value supplied where object required)",
    'opera#sourceloc': 42,
    stacktrace: "  Line 42 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "                this.undef();\n" +
        "  Line 27 of linked script file://localhost/G:/js/stacktrace.js\n" +
        "            ex = ex || this.createException();\n" +
        "  Line 18 of linked script file://localhost/G:/js/stacktrace.js: In function printStackTrace\n" +
        "        var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "  Line 4 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function bar\n" +
        "             printTrace(printStackTrace());\n" +
        "  Line 7 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function bar\n" +
        "           bar(n - 1);\n" +
        "  Line 11 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html: In function foo\n" +
        "           bar(2);\n" +
        "  Line 15 of inline#1 script in file://localhost/G:/js/test/functional/testcase1.html\n" +
        "         foo();\n" +
        ""
};

CapturedExceptions.opera_1063 = {
    message: "'this.undef' is not a function",
    stack: "<anonymous function: createException>([arguments not available])@file://localhost/G:/js/stacktrace.js:42\n" +
        "<anonymous function: run>([arguments not available])@file://localhost/G:/js/stacktrace.js:27\n" +
        "printStackTrace([arguments not available])@file://localhost/G:/js/stacktrace.js:18\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:4\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:7\n" +
        "foo([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:11\n" +
        "@file://localhost/G:/js/test/functional/testcase1.html:15",
    stacktrace: "<anonymous function: createException>([arguments not available])@file://localhost/G:/js/stacktrace.js:42\n" +
        "<anonymous function: run>([arguments not available])@file://localhost/G:/js/stacktrace.js:27\n" +
        "printStackTrace([arguments not available])@file://localhost/G:/js/stacktrace.js:18\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:4\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:7\n" +
        "foo([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:11\n" +
        "@file://localhost/G:/js/test/functional/testcase1.html:15"
};

CapturedExceptions.opera_1111 = {
    message: "'this.undef' is not a function",
    stack: "<anonymous function: createException>([arguments not available])@file://localhost/G:/js/stacktrace.js:42\n" +
        "<anonymous function: run>([arguments not available])@file://localhost/G:/js/stacktrace.js:27\n" +
        "printStackTrace([arguments not available])@file://localhost/G:/js/stacktrace.js:18\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:4\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:7\n" +
        "foo([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:11\n" +
        "@file://localhost/G:/js/test/functional/testcase1.html:15",
    stacktrace: "Error thrown at line 42, column 12 in <anonymous function: createException>() in file://localhost/G:/js/stacktrace.js:\n" +
        "    this.undef();\n" +
        "called from line 27, column 8 in <anonymous function: run>(ex) in file://localhost/G:/js/stacktrace.js:\n" +
        "    ex = ex || this.createException();\n" +
        "called from line 18, column 4 in printStackTrace(options) in file://localhost/G:/js/stacktrace.js:\n" +
        "    var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "called from line 4, column 5 in bar(n) in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    printTrace(printStackTrace());\n" +
        "called from line 7, column 4 in bar(n) in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    bar(n - 1);\n" +
        "called from line 11, column 4 in foo() in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    bar(2);\n" +
        "called from line 15, column 3 in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    foo();"
};

CapturedExceptions.opera_1151 = {
    message: "'this.undef' is not a function",
    stack: "<anonymous function: createException>([arguments not available])@file://localhost/G:/js/stacktrace.js:42\n" +
        "<anonymous function: run>([arguments not available])@file://localhost/G:/js/stacktrace.js:27\n" +
        "printStackTrace([arguments not available])@file://localhost/G:/js/stacktrace.js:18\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:4\n" +
        "bar([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:7\n" +
        "foo([arguments not available])@file://localhost/G:/js/test/functional/testcase1.html:11\n" +
        "@file://localhost/G:/js/test/functional/testcase1.html:15",
    stacktrace: "Error thrown at line 42, column 12 in <anonymous function: createException>() in file://localhost/G:/js/stacktrace.js:\n" +
        "    this.undef();\n" +
        "called from line 27, column 8 in <anonymous function: run>(ex) in file://localhost/G:/js/stacktrace.js:\n" +
        "    ex = ex || this.createException();\n" +
        "called from line 18, column 4 in printStackTrace(options) in file://localhost/G:/js/stacktrace.js:\n" +
        "    var p = new printStackTrace.implementation(), result = p.run(ex);\n" +
        "called from line 4, column 5 in bar(n) in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    printTrace(printStackTrace());\n" +
        "called from line 7, column 4 in bar(n) in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    bar(n - 1);\n" +
        "called from line 11, column 4 in foo() in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    bar(2);\n" +
        "called from line 15, column 3 in file://localhost/G:/js/test/functional/testcase1.html:\n" +
        "    foo();"
};

CapturedExceptions.chrome_15 = {
    'arguments': ["undef"],
    message: "Object #<Object> has no method 'undef'",
    stack: "TypeError: Object #<Object> has no method 'undef'\n" +
        "    at Object.createException (http://127.0.0.1:8000/js/stacktrace.js:42:18)\n" +
        "    at Object.run (http://127.0.0.1:8000/js/stacktrace.js:31:25)\n" +
        "    at printStackTrace (http://127.0.0.1:8000/js/stacktrace.js:18:62)\n" +
        "    at bar (http://127.0.0.1:8000/js/test/functional/testcase1.html:13:17)\n" +
        "    at bar (http://127.0.0.1:8000/js/test/functional/testcase1.html:16:5)\n" +
        "    at foo (http://127.0.0.1:8000/js/test/functional/testcase1.html:20:5)\n" +
        "    at http://127.0.0.1:8000/js/test/functional/testcase1.html:24:4"
};

CapturedExceptions.chrome_27 = {
    message: "Cannot call method 'undef' of null",
    name: "TypeError",
    stack: "TypeError: Cannot call method 'undef' of null\n" +
        "    at file:///E:/javascript-stacktrace/test/functional/ExceptionLab.js:4:9\n" +
        "    at createException (file:///E:/javascript-stacktrace/test/functional/ExceptionLab.js:8:5)\n" +
        "    at createException4 (file:///E:/javascript-stacktrace/test/functional/ExceptionLab.html:56:16)\n" +
        "    at dumpException4 (file:///E:/javascript-stacktrace/test/functional/ExceptionLab.html:60:23)\n" +
        "    at HTMLButtonElement.onclick (file:///E:/javascript-stacktrace/test/functional/ExceptionLab.html:83:126)"
};

CapturedExceptions.firefox_3_6 = {
    fileName: "http://127.0.0.1:8000/js/stacktrace.js",
    lineNumber: 44,
    message: "this.undef is not a function",
    name: "TypeError",
    stack: "()@http://127.0.0.1:8000/js/stacktrace.js:44\n" +
        "(null)@http://127.0.0.1:8000/js/stacktrace.js:31\n" +
        "printStackTrace()@http://127.0.0.1:8000/js/stacktrace.js:18\n" +
        "bar(1)@http://127.0.0.1:8000/js/test/functional/testcase1.html:13\n" +
        "bar(2)@http://127.0.0.1:8000/js/test/functional/testcase1.html:16\n" +
        "foo()@http://127.0.0.1:8000/js/test/functional/testcase1.html:20\n" +
        "@http://127.0.0.1:8000/js/test/functional/testcase1.html:24\n" +
        ""
};

CapturedExceptions.firefox_3_6_file = {
    fileName: "file:///home/user/js/stacktrace.js",
    lineNumber: 44,
    message: "this.undef is not a function",
    name: "TypeError",
    stack: "()@file:///home/user/js/stacktrace.js:44\n" +
        "(null)@file:///home/user/js/stacktrace.js:31\n" +
        "printStackTrace()@file:///home/user/js/stacktrace.js:18\n" +
        "bar(1)@file:///home/user/js/test/functional/testcase1.html:13\n" +
        "bar(2)@file:///home/user/js/test/functional/testcase1.html:16\n" +
        "foo()@file:///home/user/js/test/functional/testcase1.html:20\n" +
        "@file:///home/user/js/test/functional/testcase1.html:24\n" +
        ""
};

CapturedExceptions.firefox_7 = {
    fileName: "file:///G:/js/stacktrace.js",
    lineNumber: 44,
    stack: "()@file:///G:/js/stacktrace.js:44\n" +
        "(null)@file:///G:/js/stacktrace.js:31\n" +
        "printStackTrace()@file:///G:/js/stacktrace.js:18\n" +
        "bar(1)@file:///G:/js/test/functional/testcase1.html:13\n" +
        "bar(2)@file:///G:/js/test/functional/testcase1.html:16\n" +
        "foo()@file:///G:/js/test/functional/testcase1.html:20\n" +
        "@file:///G:/js/test/functional/testcase1.html:24\n" +
        ""
};

CapturedExceptions.firefox_14 = {
    message: "x is null",
    stack: "@file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html:48\n" +
        "dumpException3@file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html:52\n" +
        "onclick@file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html:1\n" +
        "",
    fileName: "file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html",
    lineNumber: 48
};

CapturedExceptions.firefox_22 = {
    message: "x is null",
    name: "TypeError",
    stack: "@file:///E:/javascript-stacktrace/test/functional/ExceptionLab.js:4\n" +
        "createException@file:///E:/javascript-stacktrace/test/functional/ExceptionLab.js:8\n" +
        "createException4@file:///E:/javascript-stacktrace/test/functional/ExceptionLab.html:56\n" +
        "dumpException4@file:///E:/javascript-stacktrace/test/functional/ExceptionLab.html:60\n" +
        "onclick@file:///E:/javascript-stacktrace/test/functional/ExceptionLab.html:1\n" +
        "",
    fileName: "file:///E:/javascript-stacktrace/test/functional/ExceptionLab.js",
    lineNumber: 4,
    columnNumber: 6
};

CapturedExceptions.safari_6 = {
    message: "'null' is not an object (evaluating 'x.undef')",
    stack: "@file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html:48\n" +
        "dumpException3@file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html:52\n" +
        "onclick@file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html:82\n" +
        "[native code]",
    line: 48,
    sourceURL: "file:///Users/eric/src/javascript-stacktrace/test/functional/ExceptionLab.html"
};

CapturedExceptions.ie_10 = {
    message: "Unable to get property 'undef' of undefined or null reference",
    stack: "TypeError: Unable to get property 'undef' of undefined or null reference\n" +
        "   at Anonymous function (http://jenkins.eriwen.com/job/stacktrace.js/ws/test/functional/ExceptionLab.html:48:13)\n" +
        "   at dumpException3 (http://jenkins.eriwen.com/job/stacktrace.js/ws/test/functional/ExceptionLab.html:46:9)\n" +
        "   at onclick (http://jenkins.eriwen.com/job/stacktrace.js/ws/test/functional/ExceptionLab.html:82:1)",
    description: "Unable to get property 'undef' of undefined or null reference",
    number: -2146823281
};
